### DISCUSSION TITLE
Multiplayer

### DISCUSSION THREAD

**Discussion by lxgr-linux** [lxgr-linux/pokete#150] [Ideas]
As I teased in some Reddit posts and was requested several times (#149), multiplayer will be the "next big thing" being added.

But the big question here is "How?". Here are some approaches I thought of:
- PvP just like with the trader
- A separate mode with multiple players playing in one world, and seeing each other, being able to fight PvP, PvE, chat etc.

What do you think/what other approaches do you think of?

**Comment by Dagonali** [comment]
This project is awesome!
Your second approach would be phenomenal!
- A separate mode with multiple players playing in one world, and seeing each other, being able to fight PvP, PvE, chat etc.    

Is there a discord for Pokete? Would be cool so people could chat there together and think of new ideas together!

**Comment by lxgr-linux** [comment]
Another very relevant problem multiplayer is how to prevent cheating.
The may purpose of a multiplayer option would be to enable the players to fight against each other, with their Poketes and items etc. they earned in sigleplayer.

But the possibility of just editing the savefile just defeats this.

So what would be possibilities of making sure savefile editing isn't possible without having a binary BLOB build in?

**Reply by AVoSI-VI** [reply]
Just spit balling, but encrypt the file basically making it unreadable?

**Reply by lxgr-linux** [reply]
An option would also be to just sign the file, but the code for that must be in the game, and therefore readable.

**Reply by AVoSI-VI** [reply]
server side validation since it will be multiplayer?

**Reply by lxgr-linux** [reply]
But then the singleplayer mode has to be online.

**Reply by AVoSI-VI** [reply]
You're right. I guess my logic would be that you download it -> it gets signed with the initial setup -> store on the server -> they connect to multiplayer sometime later down the road -> can't connect due to modified file/can connect because everything checks out

**Reply by lxgr-linux** [reply]
But the files needs to be resigned every time the games saves.
